In [1]:
import pandas as pd

In [2]:
data = pd.read_csv('movies.csv')
data

,MOVIES,YEAR,GENRE,RATING,ONE-LINE,STARS,VOTES,RunTime,Gross
0,Blood Red Sky,(2021),"\nAction, Horror, Thriller",6.1,\nA woman with a mysterious illness is forced ...,\n Director:\nPeter Thorwarth\n| \n Star...,"21,062",121.0,NaN
1,Masters of the Universe: Revelation,(2021– ),"\nAnimation, Action, Adventure",5.0,\nThe war for Eternia begins again in what may...,"\n \n Stars:\nChris Wood, \nSara...","17,870",25.0,NaN
2,The Walking Dead,(2010–2022),"\nDrama, Horror, Thriller",8.2,\nSheriff Deputy Rick Grimes wakes up from a c...,"\n \n Stars:\nAndrew Lincoln, \n...","885,805",44.0,NaN
3,Rick and Morty,(2013– ),"\nAnimation, Adventure, Comedy",9.2,\nAn animated series that follows the exploits...,"\n \n Stars:\nJustin Roiland, \n...","414,849",23.0,NaN
4,Army of Thieves,(2021),"\nAction, Crime, Horror",NaN,"\nA prequel, set before the events of Army of ...",\n Director:\nMatthias Schweighöfer\n| \n ...,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...
9994,The Imperfects,(2021– ),"\nAdventure, Drama, Fantasy",NaN,\nAdd a Plot\n,\n \n Stars:\nMorgan Taylor Camp...,NaN,NaN,NaN
9995,Arcane,(2021– ),"\nAnimation, Action, Adventure",NaN,\nAdd a Plot\n,\n,NaN,NaN,NaN
9996,Heart of Invictus,(2022– ),"\nDocumentary, Sport",NaN,\nAdd a Plot\n,\n Director:\nOrlando von Einsiedel\n| \n ...,NaN,NaN,NaN
9997,The Imperfects,(2021– ),"\nAdventure, Drama, Fantasy",NaN,\nAdd a Plot\n,\n Director:\nJovanka Vuckovic\n| \n Sta...,NaN,NaN,NaN


In [3]:
data.columns

Index(['MOVIES', 'YEAR', 'GENRE', 'RATING', 'ONE-LINE', 'STARS', 'VOTES',
       'RunTime', 'Gross'],
      dtype='object')

# Gathered insights 

- [] Movies handle extra spaces and duplicates 
- [] year handle missing values which are 644 and make sure 200x - change to 2xxx and in interger dtype
- [] genre handle special chracter andand missing value of 80
- [] Rating it has 1820 missing values handle them make sure it is from 0 to 10 only and in interger dtype
- [] One line make sure it has no extra spaces and unwanted character and in standardized format 
- [] votes it has many missing values which are around 1820 make sure they are in postive numbers and in interger dtype
- [] Runtime it has 2958 missing values so better handle them and it should be non in negative 
- [] Fross it is better remove tha columsn as it has most of the missing values 

 - []MOVIES
    Strip extra spaces. Drop the 431 exact duplicate rows.
- []    YEAR
Pull just the year number out using regex, since the raw values look like (2021) or (2010–2022). Watch out for entries starting with (I) or (II) — don't let your regex break on those.

- [] GENRE
Strip the leading/trailing whitespace and newline characters. Handle the 80 missing values however you like (drop or fill with "Unknown").

- [] RATING
Keep it as decimal numbers (don't convert to whole integers — you'd lose real precision like 8.7 becoming 9). For the 1,820 missing values: don't drop those rows. Either fill them in with the median rating, or just leave them blank — either is fine for now.

- [] ONE-LINE
Strip the leading \n and extra whitespace.

- [] STARS
Split this into two separate columns — Director and Stars — since right now they're jammed together as one messy text blob with labels and newlines.

- [] VOTES
Remove the commas from numbers like 21,062 and convert to plain integers. Same as Rating — don't drop the 1,820 missing rows, just fill or leave them blank.

- [] RUNTIME
Before deciding anything, check if the really long runtimes (300+ minutes) belong to TV shows vs movies — they likely mean different things. Then fill in the missing values with the median.

- [] GROSS
Just drop this whole column — over 95% of it is missing.

In [7]:
data.isna().sum()

MOVIES         0
YEAR         644
GENRE         80
RATING      1820
ONE-LINE       0
STARS          0
VOTES       1820
RunTime     2958
Gross       9539
dtype: int64

In [8]:
df = data.copy()

In [9]:
df.head()

,MOVIES,YEAR,GENRE,RATING,ONE-LINE,STARS,VOTES,RunTime,Gross
0,Blood Red Sky,(2021),"\nAction, Horror, Thriller",6.1,\nA woman with a mysterious illness is forced ...,\n Director:\nPeter Thorwarth\n| \n Star...,"21,062",121.0,NaN
1,Masters of the Universe: Revelation,(2021– ),"\nAnimation, Action, Adventure",5.0,\nThe war for Eternia begins again in what may...,"\n \n Stars:\nChris Wood, \nSara...","17,870",25.0,NaN
2,The Walking Dead,(2010–2022),"\nDrama, Horror, Thriller",8.2,\nSheriff Deputy Rick Grimes wakes up from a c...,"\n \n Stars:\nAndrew Lincoln, \n...","885,805",44.0,NaN
3,Rick and Morty,(2013– ),"\nAnimation, Adventure, Comedy",9.2,\nAn animated series that follows the exploits...,"\n \n Stars:\nJustin Roiland, \n...","414,849",23.0,NaN
4,Army of Thieves,(2021),"\nAction, Crime, Horror",NaN,"\nA prequel, set before the events of Army of ...",\n Director:\nMatthias Schweighöfer\n| \n ...,NaN,NaN,NaN


In [10]:
df.duplicated().sum()

np.int64(431)

In [12]:
df = df.drop_duplicates()

In [14]:
df.describe()

,RATING,RunTime
count,8168.000000,7008.000000
mean,6.919699,68.874144
std,1.219970,47.278890
min,1.100000,1.000000
25%,6.200000,37.000000
50%,7.100000,60.000000
75%,7.800000,95.000000
max,9.900000,853.000000


In [16]:
df.columns

Index(['MOVIES', 'YEAR', 'GENRE', 'RATING', 'ONE-LINE', 'STARS', 'VOTES',
       'RunTime', 'Gross'],
      dtype='object')